server_fn/src/error.rs

```diff
@@ -505,34 +526,54 @@ impl<E> From<ServerFnUrlError<ServerFnError<E>>> for ServerFnError<E> {
     }
 }
 
-#[derive(Debug)]
+#[derive(Debug, thiserror::Error)]
 #[doc(hidden)]
 /// Only used instantly only when a framework needs E: Error.
 pub struct ServerFnErrorWrapper<E: FromServerFnError>(pub E);
 
 impl<E: FromServerFnError> Display for ServerFnErrorWrapper<E> {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        write!(f, "{}", self.0.ser())
+        write!(
+            f,
+            "{}",
+            <E::Encoder as FormatType>::into_encoded_string(self.0.ser())
+        )
     }
 }
 
-impl<E: FromServerFnError> std::error::Error for ServerFnErrorWrapper<E> {
-    fn source(&self) -> Option<&(dyn std::error::Error + 'static)> {
-        None
+impl<E: FromServerFnError> FromStr for ServerFnErrorWrapper<E> {
+    type Err = base64::DecodeError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let bytes =
+            <E::Encoder as FormatType>::from_encoded_string(s).map_err(|e| {
+                E::from_server_fn_error(ServerFnErrorErr::Deserialization(
+                    e.to_string(),
+                ))
+            });
+        let bytes = match bytes {
+            Ok(bytes) => bytes,
+            Err(err) => return Ok(Self(err)),
+        };
+        let err = E::de(bytes);
+        Ok(Self(err))
     }
 }
 
 /// A trait for types that can be returned from a server function.
 pub trait FromServerFnError:
-    std::fmt::Debug + Serialize + DeserializeOwned + 'static
+    std::fmt::Debug + Sized + Display + 'static
```

Comment:
Did we really have to go back to the `Display` bound? I was so happy when I learned that 0.8 wasn't going to require the custom errors to have `Display` anymore, and now this obnoxious requirement is back? 😩